pkg/component/kubernetes/controllermanager/controllermanager.go

```diff
@@ -575,6 +578,37 @@ func (k *kubeControllerManager) Deploy(ctx context.Context) error {
 		}
 	}
 
+	if k.values.NamePrefix != "" {
+		serviceMonitor := k.emptyServiceMonitor()
+		if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, k.seedClient.Client(), serviceMonitor, func() error {
+			serviceMonitor.Labels = monitoringutils.Labels(garden.Label)
+			serviceMonitor.Spec = monitoringv1.ServiceMonitorSpec{
+				Selector: metav1.LabelSelector{MatchLabels: getLabels()},
+				Endpoints: []monitoringv1.Endpoint{{
+					Port:      portNameMetrics,
+					Scheme:    "https",
+					TLSConfig: &monitoringv1.TLSConfig{SafeTLSConfig: monitoringv1.SafeTLSConfig{InsecureSkipVerify: true}},
```

Comment:
When enforcing CA verification, it fails with

```
Get "https://10.1.254.86:10257/metrics": tls: failed to verify certificate: x509: cannot validate certificate for 10.1.254.86 because it doesn't contain any IP SANs
```

Perhaps it makes sense to rework this separately (if even possible) to be able to enable CA verification. cc @istvanballok